### ISSUE TITLE
Panic when upgrading from v0.7.10 to any newer version

### ISSUE THREAD

**Issue by lafriks** [goccy/go-json#324]
code panics while marshaling `req` variable:

```go
type OtherStruct struct {
	FieldA *string `json:"fieldA,omitempty"`
	FieldB *string `json:"fieldB,omitempty"`
	FieldC *bool `json:"fieldC"`
	FieldD []string `json:"fieldD,omitempty"`
}

	req := &struct {
		Code string `json:"code"`
		*OtherStruct
	}{
		OtherStruct: &OtherStruct{},
	}

	var stringVar = "Test Field"
	req.Code = "Test"
	req.OtherStruct.FieldB = &stringVar
```

**Comment by goccy** [comment]
Thank you for reporting ! I'll fix this problem with #326

**Comment by lafriks** [comment]
Can confirm that my issue is resolved with latest version. Thanks for fast response and great library!

**Comment by goccy** [comment]
Thank you for confirmation ! I released this fixing as v0.9.4
